datafusion/common/src/config.rs

```diff
@@ -1682,28 +1690,15 @@ impl Display for FormatOptions {
     }
 }
 
-impl From<FileType> for FormatOptions {
-    fn from(value: FileType) -> Self {
-        match value {
-            FileType::ARROW => FormatOptions::ARROW,
-            FileType::AVRO => FormatOptions::AVRO,
-            #[cfg(feature = "parquet")]
```

Comment:
it is certainly nice to avoid many of these `            #[cfg(feature = "parquet")]` that were sprinkled throughout the code